cmd/gateway/s3/gateway-s3.go

```diff
@@ -474,6 +482,14 @@ func (l *s3Objects) CopyObject(ctx context.Context, srcBucket string, srcObject
 	// So preserve it by adding "REPLACE" directive to save all the metadata set by CopyObject API.
 	srcInfo.UserDefined["x-amz-metadata-directive"] = "REPLACE"
 	srcInfo.UserDefined["x-amz-copy-source-if-match"] = srcInfo.ETag
+	// See comment above. Tags will be applied on destination object.
+	if !minio.IsStringEqual(srcInfo.UserDefined["X-Amz-Tagging"], "") {
+		if minio.IsStringEqual(srcInfo.UserDefined["X-Amz-Tagging-Directive"], "") {
+			srcInfo.UserDefined["x-amz-tagging-directive"] = "REPLACE"
+		}
+		srcInfo.UserTags = srcInfo.UserDefined["X-Amz-Tagging"]
```

Comment:
Is the tags getting used anywhere?

Reply:
without `REPLACE` does it make sense to have the `tags` sent over from the client?

Reply:
In this case of CopyObject() UserTags struct member are being used in minio-go. I will remove the condition around REPLACE.

Reply:
I've removed setting of UserTags. Testing showed it to work with both S3 & minio server. REPLACE directive is needed by the functionality.